backend/src/test/scala/cromwell/backend/io/WorkflowPathsSpec.scala

```diff
@@ -52,12 +72,23 @@ class WorkflowPathsSpec extends FlatSpec with Matchers with BackendSpec {
 
     subWd.breadCrumbs returns List(BackendJobBreadCrumb(rootWorkflow, rootWorkflowId, jobKey))
     subWd.id returns subWorkflowId
-    
+
     val workflowPaths = new WorkflowPathsWithDocker(subWd, backendConfig)
+    val expectedRoot = root.getOrElse(WorkflowPaths.defaultExecutionRootString)
+    val expectedDockerRoot = dockerRoot.getOrElse(WorkflowPathsWithDocker.defaultDockerRoot)
+
     workflowPaths.workflowRoot.pathAsString shouldBe
       DefaultPathBuilder.get(
-        s"local-cromwell-executions/rootWorkflow/$rootWorkflowId/call-call1/shard-1/attempt-2/subWorkflow/$subWorkflowId"
+        s"$expectedRoot/rootWorkflow/$rootWorkflowId/call-call1/shard-1/attempt-2/subWorkflow/$subWorkflowId"
       ).toAbsolutePath.pathAsString
-    workflowPaths.dockerWorkflowRoot.pathAsString shouldBe s"/cromwell-executions/rootWorkflow/$rootWorkflowId/call-call1/shard-1/attempt-2/subWorkflow/$subWorkflowId"
+    workflowPaths.dockerWorkflowRoot.pathAsString shouldBe s"$expectedDockerRoot/rootWorkflow/$rootWorkflowId/call-call1/shard-1/attempt-2/subWorkflow/$subWorkflowId"
+  }
+
+  "WorkflowPaths" should "provide correct paths for a workflow" in {
+    testConfigs.foreach(config => testWorkflowPaths(config.root, config.dockerRoot))
```

Comment:
Thanks @cjllanwarne . In our scala tool suite we use [TestNG](http://testng.org/doc/index.html). Which has nice built-in support for fixtures using a `DataProvider`. So I was unsure how to do this in native scalatest.
A DataProvider looks like [this](https://github.com/biopet/bamstats/blob/4d324036eafb1788e473b1127e8dae33bdf3f1bc/src/test/scala/nl/biopet/tools/bamstats/schema/SchemaTest.scala#L44). There is a test that uses it directly below. It will automatically separate out test results for each of the provided data structures. So maybe this can be of use for cromwell development?